### PR TITLE
[5.2] Add getBroker() and getGuard()

### LINKED ISSUE #11695: No Guard Instance for Password Reset? Laravel 5.2.0
I have successfully tried 2 different Database table Login for 2 different area. Then I thought it is a great scalability option for Laravel. But then when I tried Resetting Password for Login Credentials other than default users table. It stopped. It only runs query on the users table no matter where you are.

In L-5.2.0 specifying other guard instance is introduced for Auth Facade. eg.
`Auth::guard("GuardInstanceName")`
But I think it is required for Password Facade too. Something like:
`Password::guard("GuardInstanceName")->reset(Credentials, Callback);`

### PR DESCRIPTION
**Pull request by hiendv**
Resolve #11695

### CODE REVIEW

File: src/Illuminate/Foundation/Auth/ResetsPasswords.php [hunk 1]
```diff
@@ -12,6 +12,25 @@
     use RedirectsUsers;
 
     /**
+     * Get the broker to be used during resetting the password
+     * @return string|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ResetsPasswords.php]: missing newline

File: src/Illuminate/Foundation/Auth/ResetsPasswords.php [hunk 2]
```diff
@@ -12,6 +12,25 @@
     use RedirectsUsers;
 
     /**
+     * Get the broker to be used during resetting the password
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ResetsPasswords.php]: also, missing a fullstop as picked up by styleci

**Comment by hiendv** [comment]
@GrahamCampbell Sorry again! It was a bit rushed.

**Comment by hiendv** [comment]
@taylorotwell The description has been updated, just in case you ask for the use case ;)

**Pull request merged**